app/actions/remote/entry/common.ts

```diff
@@ -210,6 +212,15 @@ export const deferredAppEntryActions = async (
 
     fetchAllTeams(serverUrl);
     updateAllUsersSince(serverUrl, since);
+
+    if (teamData.teams?.length && teamData.memberships?.length) {
+        /* eslint-disable no-await-in-loop */
+        for (const team of teamData.teams) {
+            // need to await here since GM/DM threads in different teams overlap
+            await fetchNewThreads(serverUrl, team.id, false);
+        }
+        /* eslint-enable no-await-in-loop */
```

Comment:
```suggestion
        for await (const team of teamData.teams) {
            // need to await here since GM/DM threads in different teams overlap
            await fetchNewThreads(serverUrl, team.id, false);
        }
```

Reply:
Also let's talk about this as I believe we can make further improvements

Reply:
I've made some more changes (not pushed), essentially 2 functions:
- fetchNewThreadsForTeam
- fetchNewThreadsForMultipleTeams

but let's discuss in sync